components/base-node.js

```diff
@@ -65,7 +79,32 @@ module.exports = {
        ar.push(item);
      }
      return ar;
-   } 
+   },
```

Comment:
That is a simple helper function that James was using in multiple places to dynamically build up his incoming port data.    Rather than duplicate it in multiple components, I centralized it here. Look at rdf-query and rdf-update for good examples of the use case.